src/test/java/uk/ac/ebi/ampt2d/accession/AccessioningServiceTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.ampt2d.accession;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+public class AccessioningServiceTest {
+
+    private  AccessioningService<String> service;
+
+    @Before
+    public void setUp() throws Exception {
+        AccessionGenerator<String> generator = new AccessionGenerator<String>() {
+            private static final String PREFIX = "ACC";
+
+            private int counter = 0;
+
+            @Override
+            public Map<String, String> get(Set<String> objects) {
+                Map<String, String> accessions = new HashMap<>();
+                for (String object : objects) {
+                    accessions.put(object, PREFIX + counter++);
+                }
+                return accessions;
+            }
+        };
+
+        AccessionRepository<String> repository = new AccessionRepository<String>() {
+            private Map<String, String> storage = new HashMap<>();
+
+            @Override
+            public Map<String, String> get(List<String> objects) {
+                Map<String, String> accessions = new HashMap<>();
+                for (String object : objects) {
+                    if (storage.containsKey(object)) {
+                        accessions.put(object, storage.get(object));
+                    }
+                }
+                return accessions;
+            }
+
+            @Override
+            public void add(Map<String, String> accessions) {
+                storage.putAll(accessions);
+            }
+        };
+
+        service = new AccessioningService<>(repository, generator);
+    }
+
+    @Test
+    public void everyNewObjectReceiveOneAccession() throws Exception {
+        List<String> newObjects = Arrays.asList("Object 1", "Object 2", "Object 3");
+        Map<String, String> accessions = service.getAccessions(newObjects);
+
+        assertTrue(newObjects.stream().allMatch(object -> accessions.get(object) != null));
```

Comment:
I'm not sure about this, it could be a bit more clear with a loop and `assertNotNull` used for each element.